File: frontend/src/types/decision-alert.ts
export type DecisionAlertItem = {
  id: number | null;
  ticker: string;
  display_name: string;
  topic_name: string | null;
  alert_type: string;
  priority: string;
  title: string;
  body: string;
  next_action: string;
  action: string;
  confidence: number;
  source: string;
  reasons: string[];
  dedupe_key?: string | null;
  read_at?: string | null;
  dismissed_at?: string | null;
  created_at?: string | null;
  updated_at?: string | null;
};

export type DecisionAlertSummary = {
  generated_at: string;
  available: boolean;
  items: DecisionAlertItem[];
  count: number;
  empty_message: string | null;
};

export type DecisionAlertList = {
  generated_at: string;
  unread_count: number;
  items: DecisionAlertItem[];
  count: number;
};

export type DecisionAlertReadAllResult = {
  generated_at: string;
  updated_count: number;
};

File: python/valuecell/server/api/schemas/decision_alert.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class DecisionAlertItem(BaseModel):
    id: int | None = Field(None, description="Alert id")
    ticker: str = Field(..., description="Alert ticker")
    display_name: str = Field(..., description="Display name")
    topic_name: str | None = Field(None, description="Related topic name")
    alert_type: str = Field(..., description="Alert type")
    priority: str = Field(..., description="Priority level")
    title: str = Field(..., description="Alert title")
    body: str = Field(..., description="Alert body")
    next_action: str = Field(..., description="Next observation action")
    action: str = Field(..., description="Source action")
    confidence: int = Field(..., description="Confidence score")
    source: str = Field(..., description="Alert source")
    reasons: list[str] = Field(default_factory=list, description="Alert reasons")
    dedupe_key: str | None = Field(None, description="Dedupe key")
    read_at: datetime | None = Field(None, description="Read timestamp")
    dismissed_at: datetime | None = Field(None, description="Dismissed timestamp")
    created_at: datetime | None = Field(None, description="Created timestamp")
    updated_at: datetime | None = Field(None, description="Updated timestamp")


class DecisionAlertSummaryData(BaseModel):
    generated_at: datetime = Field(..., description="Generation time")
    available: bool = Field(..., description="Whether alerts are available")
    items: list[DecisionAlertItem] = Field(default_factory=list)
    count: int = Field(..., description="Alert count")
    empty_message: str | None = Field(None, description="Empty fallback message")


class DecisionAlertListData(BaseModel):
    generated_at: datetime = Field(..., description="Generation time")
    unread_count: int = Field(..., description="Unread alert count")
    items: list[DecisionAlertItem] = Field(default_factory=list)
    count: int = Field(..., description="Returned alert count")


class DecisionAlertReadAllData(BaseModel):
    generated_at: datetime = Field(..., description="Operation time")
    updated_count: int = Field(..., description="Updated alert count")
